config/docker/server/startServer.sh.mustache

```diff
@@ -1,17 +1,92 @@
 #!/bin/bash
 set -e
 
+function try()
+{
+    [[ $- = *e* ]]; SAVED_OPT_E=$?
+    set +e
+}
+
+function throw()
+{
+    exit $1
+}
+
+function catch()
+{
+    export ex_code=$?
+    (( $SAVED_OPT_E )) && set +e
+    return $ex_code
+}
+
 name=$1
 mode=$2
+waitForOther=$3
+application="server"
+theOtherApplication="broker"
+config="./$application-config"
 
-echo "RUNNING startServer.sh $name $mode"
+echo "RUNNING $application $name $mode"
+export LD_LIBRARY_PATH={{{catapultAppFolder}}}/lib:{{{catapultAppFolder}}}/deps
 
 ulimit -c unlimited
 
+rm -f "{{{dataDirectory}}}/$application.started"
+rm -f "{{{dataDirectory}}}/$application-recovery.started"
+
 if [ "$mode" == "DEBUG" ]; then
   echo "Setting up core dump..."
   mkdir -p ./logs
-  echo './logs/server.%e.%t' >/proc/sys/kernel/core_pattern
+  echo "./logs/$application.%e.%t" >/proc/sys/kernel/core_pattern
 fi
 
-exec env LD_LIBRARY_PATH={{{catapultAppFolder}}}/lib:{{{catapultAppFolder}}}/deps {{{catapultAppFolder}}}/bin/catapult.server ./userconfig
+if [ "$waitForOther" == "true" ]; then
```

Comment:
The wait should be after the recovery is run. If the other application has started the recovery will fail.

Reply:
Thanks, I'll move. it.
Question. Do you have a strategy about how to stop one service when the other one fails? If both catapult.broker and catapult.,server are running, and one stops. How to kill the other and clean up the flags so both services start (and recover) from scratch.

I'm thinking about pooling for the "theOtherApplication.started". It this file is removed (the other processed stoped), kill the current process. Do you have something better? Another alternative is using compose's health checks. But I don't want to ping the server.

Reply:
Push some changes that will do the automatic recovery the api node.  Take a look....It works but needs some cleanup.

Reply:
Thanks @Wayonb. I looks really nice.

I've merged the 2 scripts together into start.sh. It's not a mustache template anymore. I've added -f to the `rm` commands. The start.sh could be used potentially for non-docker-compose node admins. 

Could you validate https://github.com/nemtech/symbol-bootstrap/blob/broker_server_config/config/docker/server/start.sh